### PR TITLE
feat(transport): add mutex for sync tcpsocket

### PR DESCRIPTION
**Pull request by RomanAyupov89**

### CODE REVIEW

File: src/marionette/transport.cr [hunk 1]
```diff
@@ -63,7 +66,7 @@ module Marionette
           end
         end
       end
-
+      Log.trace { "(#{@id}) Marionette receive #{raw}" }
```
Comment (dammer) [inline src/marionette/transport.cr]: It i can be really big, can we limit that by some appropriate size?

Comment (dammer) [reply src/marionette/transport.cr]: If it is really important to debugging just close conversation as resolved.

**Review by dammer — APPROVED** [review]

File: src/marionette/transport.cr (hunk 1 above)
Comment (RomanAyupov89) [reply src/marionette/transport.cr]: I specially set that this log level trace. Our puppet logs are not included. I have not noticed large logs at home. I think it's worth leaving. if it interferes with local launches then we will delete) @dammer , what do you say?

Comment (alonchin) [reply src/marionette/transport.cr]: agree with Damir, not sure why we need all this in log. I think log about 100-200 bytes would be enough.

**Review by alonchin — APPROVED** [review]
does mutex fix the issue or it is just for testing?

**Comment by RomanAyupov89** [comment]
> does mutex fix the issue or it is just for testing?

i think fixed. i will update jira ticket

File: src/marionette/transport.cr (hunk 1 above)
Comment (RomanAyupov89) [reply src/marionette/transport.cr]: if you insist) Fixed

**Review by alonchin — APPROVED** [review]

**Pull request merged**